src/annotator/test/guest-test.js

```diff
@@ -259,6 +259,24 @@ describe('Guest', () => {
 
         assert.notCalled(fakeIntegration.fitSideBySide);
       });
+
+      it('emits a "hypothesis:layoutchange" DOM event', done => {
+        const guest = createGuest();
+        const dummyLayout = {
+          expanded: true,
+          width: 100,
+          height: 300,
+          toolbarWidth: 10,
+        };
+
+        guest.element.addEventListener('hypothesis:layoutchange', e => {
+          assert.equal(e.detail.sidebarLayout, dummyLayout);
+          // This ensures the test will timeout if this event is not emitted
+          done();
```

Comment:
I know using `done` is a bit "ugly" and usually not recommended, but I think in this case it is simple enough. Without it, if the event is not triggered, it won't make the test fail.

Using your suggestion, it could look like this:

```ts
// ...

let resolvePromise;
const promise = new Promise(resolve => {
  resolvePromise = resolve;
});

guest.element.addEventListener('hypothesis:layoutchange', e => {
  assert.equal(e.detail.sidebarLayout, dummyLayout);
  // This ensures the test will timeout if this event is not emitted
  resolvePromise();
});

emitHostEvent('sidebarLayoutChanged', dummyLayout);

await promise;
```

It works, but I'm not sure if I like it better than `done` 😅 

Unless you were talking of something else and I didn't get your suggestion.

Reply:
Actually I just saw your other comment https://github.com/hypothesis/client/pull/5555/files#r1236704084.

That makes much more sense. I'll change it.